templates/usr/local/sbin/simp_generate_types.epp

```diff
@@ -0,0 +1,104 @@
+#!/opt/puppetlabs/puppet/bin/ruby
+
+require 'ostruct'
+require 'optparse'
+require 'syslog/logger'
+
+log = Syslog::Logger.new(File.basename(__FILE__))
```

Comment:
Should you set the syslog facility?

Reply:
This works as expected and the facility will be inherited by whatever is calling it.

Reply:
Don't understand 'facility will be inherited'. I understand that Syslog::Logger is a singleton.  However, whenever I have used Syslog::Logger I have explicitly set the facility and not relied on a default value.